Emscripten-build-instructions.md

```diff
@@ -0,0 +1,167 @@
+# Wasm Build Instructions
+
+## CppInterOp Wasm Build Instructions
+
+This document first starts with the instructions on how to build a wasm  
+build of CppInterOp. Before we start it should be noted that  
+unlike the non wasm version of CppInterOp we currently only  
+support the Clang-REPL backend using llvm>19 for osx and Linux.  
+We will first make folder to build our wasm build of CppInterOp.  
+This can be done by executing the following command
+
+```bash
+mkdir CppInterOp-wasm
+```
+
+Now move into this directory using the following command
+
+```bash
+cd ./CppInterOp-wasm
+```
+
+To create a wasm build of CppInterOp we make use of the emsdk  
+toolchain. This can be installed by executing (we only currently  
+support version 3.1.45)
+```bash
+git clone https://github.com/emscripten-core/emsdk.git
+cd emsdk
+./emsdk install  3.1.45
+```
+
+Go back upto the top level build directory using
+
+```bash
+cd ../..
+```
+
+and activate the emsdk environment
+```bash
+./emsdk/emsdk activate 3.1.45
+source ./emsdk/emsdk_env.sh
+```bash
+
+Now clone the 19.x release of the LLVM project repository 
+and CppInterOp
+
+```bash
+git clone --depth=1 --branch release/19.x https://github.com/llvm/llvm-project.git
+git clone --depth=1 https://github.com/compiler-research/CppInterOp.git
+```
+
+Now move into the cloned llvm-project folder and apply  
+a required patch
+
+```bash
+cd ./llvm-project/
+git apply -v ../CppInterOp/patches/llvm/emscripten-clang${{ matrix.clang-runtime }}-*.patch
+```
+
+We are now in a position to build an emscripten build of llvm by executing the following
+```bash
+mkdir build
+cd build
+emcmake cmake -DCMAKE_BUILD_TYPE=Release \
+                        -DLLVM_HOST_TRIPLE=wasm32-unknown-emscripten \
+                        -DLLVM_ENABLE_ASSERTIONS=ON                        \
+                        -DLLVM_TARGETS_TO_BUILD="WebAssembly" \
+                        -DLLVM_ENABLE_LIBEDIT=OFF \
+                        -DLLVM_ENABLE_PROJECTS="clang;lld" \
+                        -DLLVM_ENABLE_ZSTD=OFF \
+                        -DLLVM_ENABLE_LIBXML2=OFF \
+                        -DCLANG_ENABLE_STATIC_ANALYZER=OFF \
+                        -DCLANG_ENABLE_ARCMT=OFF \
+                        -DCLANG_ENABLE_BOOTSTRAP=OFF \
+                        -DCMAKE_CXX_FLAGS="-Dwait4=__syscall_wait4" \
+                        ../llvm
+emmake make clang -j $(nproc --all)
+emmake make clang-repl -j $(nproc --all)
+emmake make lld -j $(nproc --all)
+```
+
+Once this finishes building we need to take note of where we built our  
+llvm build. This can be done by executing the following
+
+```bash
+export LLVM_BUILD_DIR=$PWD
+```
```

Comment:
This is okay but in the same place ....
**I would really like to make it as easy as possible for the user to try out**

Building LLVM against emscripten would itself take greater an 1 hour. 
Now that we're using the same configs as emscripten-forge, I would just do 
1) micromamba create an environment
2) micromamba install llvm -c https://repo.mamba.pm/emscripten-forge
3) use env prefix as `llvm_build_dir`, so basically LLVM_DIR = prefix/lib/cmake/llvm 

This would get the above task done in 2 mins. 

But as you've done ... for the curious user, we also provide a way to build llvm so that they can play around with configs if interested.

Reply:
We are not quite the same (not just yet anyway 😉). These are there to provide the imquistitive user too in case they want to play with some different flags. Gives them a starting point. These instructions are for developers as well as users